tfe/resource_tfe_team_project_access.go

```diff
@@ -166,3 +436,19 @@ func resourceTFETeamProjectAccessDelete(ctx context.Context, d *schema.ResourceD
 
 	return nil
 }
+
+// You cannot set custom permissions when access level is not "custom"
+func checkForCustomPermissions(_ context.Context, d *schema.ResourceDiff, meta interface{}) error {
```

Comment:
Sigh. This doesn't work for Updates through config.
If you remove the project_access & workspace_access sections from the tf config,
using either GetOk or even GetChange -- these errors still trip. I also tried instead of "project_access", "project_access.0" etc.

GetChange is supposed to return an old value and a new value. Using prints, in the provider here and running an apply that had removed both sections -- the old and value remain the same and remain the old value.

GetChange does do what is expected if you update the permission values themselves -- just not when you remove the payload entirely which makes no sense to me.

I also tried removing it from here, and raising errors in the Create/Update methods and had the same issue in the Update method even though that's using the *schema.ResourceData and not *schema.ResourceDiff.

Reply:
Woot!! Phew!!
using the GetRawConfing... fixes the issue 😅